### PR TITLE
US123999 [AQ] Export Quiz Details Table to CSV

### PR DESCRIPTION
**Pull request by Vitalii-Misechko**
[US123999](https://rally1.rallydev.com/#/23525809118d/iterationstatus?detail=%2Fuserstory%2F483492691144&sharedViewId=328666453148&view=dc9e7043-1cf6-486d-a4aa-3a212976e789&fdp=true?fdp=true): [AQ] Export Quiz Details Table to CSV

Outside the scope of this PR:
> Warning/pop-up to appear if download will return more than limit

* Functional Testing 
  * [x] Browsers (Chrome, FF, Edge)
  * [x] Responsive
  * [x] Test cases
    * [x] Click on  `Export to CSV` generates and downloads `Assessment_Quality_Dashboard.csv`
    * [x]  `Export to CSV` generates files that include filtered records
* Security, devops, perf: N/A
* UX and docs
  * [x] Demo to Kevin

### CODE REVIEW

**Comment by Vitalii-Misechko** [comment]
FYI @Brightspace/draco-vis

File: pages/assessment-quality-dashboard-view.js [hunk 1]
```diff
@@ -282,6 +290,36 @@ class AssessmentQualityDashboardView extends SkeletonMixin(Localizer(MobxLitElem
 		window.open(helpLink, '_blank');
 	}
 
+	get dataForExport() {
+		const formatQuiz = (quiz) => `${quiz.quizName} ${this.localize('dashboard:quizDetailsQuizId')} ${quiz.quizId} ${this.localize('dashboard:quizDetailsCourseNameTemplate', { courseName: quiz.courseName, courseId: quiz.courseId })}`;
```
Comment (bemailloux) [inline pages/assessment-quality-dashboard-view.js:294]: Should quiz ID / quiz Name / course ID / course name all be in different columns? (check with Kevin)

Comment (Vitalii-Misechko) [reply pages/assessment-quality-dashboard-view.js]: Fixed. Kevin prefers separate columns

File: model/exportData.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import { toJS } from 'mobx';
+
+export class ExportData {
+	static userDataToCsv(fileName, data, headers) {
+		return this.exportToCsv(fileName, toJS(data), headers);
+	}
+
+	static exportToCsv(fileName, formattedData, headers) {
+		// exportFromJSON's built-in filename formatter calls `replace` instead of `replaceAll`
+		const safeFileName = fileName.replaceAll(' ', '_');
```
Comment (bemailloux) [inline model/exportData.js:11]: QQ: does this (and more generally, the export itself) work in languages with non-ASCII characters?

**Review by bemailloux — APPROVED** [review]

**Review by NicholasHallman — APPROVED** [review]

File: model/exportData.js (hunk 2 above)
Comment (Vitalii-Misechko) [reply model/exportData.js:11]: Yes, it does.

image
image

**Pull request merged**